test/models/content_view_test.rb

```diff
@@ -391,15 +391,18 @@ def test_docker_repo_container_names
       repo2_cv = build(:docker_repository, product: product, content_view_version: @organization.default_content_view.versions.first, library_instance_id: repo2_lib.id)
       version2 = create(:katello_content_view_version, :content_view => view2, :repositories => [repo2_cv])
 
-      composite.update(component_ids: [version1.id])
+      composite.update(component_ids: [version1.id, version2.id])
       assert composite.valid?
-      @dev.registry_name_pattern = "abcdef"
       assert composite.check_docker_repository_names!([@dev])
 
-      composite.update(component_ids: [version1.id, version2.id])
-      assert composite.valid?
-      assert_raises(RuntimeError) do
```

Comment:
This test used to verify that `check_docker_repository_names!` would error out if there is a clash between the names in an environment. Since I don't see any other test doing that anymore, I think a new test will need to be created that does just that.

It might be worth just splitting your new assertions here into a new test and remove the extra bits from `test_docker_repo_container_names` that you don't need. Then you could return `test_docker_repo_container_names` to doing what it was before.

Reply:
To me, it seems like the tail end of `test_docker_promote` handles that? I could be incorrect though. I can absolutely add/repair this if you think it's the correct thing to do.

Reply:
Looks like the original test fails because the repositories do not have any environment. If you add environments to the repositories that would be published into the composite, the test should succeed.

Reply:
> To me, it seems like the tail end of `test_docker_promote` handles that? I could be incorrect though. I can absolutely add/repair this if you think it's the correct thing to do.

`test_docker_promote` tests that `check_docker_repository_names!` succeeds, but it doesn't test that it fails.

Reply:
I believe I fixed this test but I'd love a second pair of eyes on this. I couldn't find a way to change the container repo name after creation, so I ended up just testing for an invalid record error on the attempted creation of the second identical cv.

Reply:
Because the creation fails on the creating a cv step, there are no content view versions in use here. That said, I believe it still tests whether two docker repos will error out when they share a container repository name.